cmd/policy_publish.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/pulumi/pulumi/pkg/engine"
+
+	"github.com/pulumi/pulumi/pkg/backend"
+
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi/pkg/backend/display"
+	"github.com/pulumi/pulumi/pkg/backend/httpstate"
+	"github.com/pulumi/pulumi/pkg/util/cmdutil"
+	"github.com/pulumi/pulumi/pkg/workspace"
+	"github.com/spf13/cobra"
+)
+
+func newPolicyPublishCmd() *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:   "publish <orgName>/<policyPackName>",
+		Args:  cmdutil.ExactArgs(1),
+		Short: "Publish resource policies to the Pulumi service",
+		Long:  "Publish resource policies to the Pulumi service",
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			//
+			// Obtain current PolicyPack, tied to the Pulumi service backend.
+			//
+
+			policyPack, err := requirePolicyPack(args[0])
+			if err != nil {
+				return err
+			}
+
+			//
+			// Load metadata about the current project.
+			//
+
+			// TODO: `readPolicyProject` instead?
+			proj, root, err := readProject()
```

Comment:
Fine for now, but do you think long term, do you think we could get rid of the ideal of a "project" here?  I can't imagine what information we'd store in the `Pulumi.yaml` file?  I think the only thing we could care about is the language and root, but maybe we could just say: "You have to run this from a directory with a `package.json` in it?"

Reply:
We're using this for the language metadata. Happy to hear other ways to get this working without it.